CI: xfail sparse warning

this is failing builds occasionally. not sure if its a deeper problem. can just xfail for now and investigate.

https://travis-ci.org/pandas-dev/pandas/jobs/641752516

```
=================================== FAILURES ===================================
__________________________ test_legacy_sparse_warning __________________________
[gw0] linux -- Python 3.6.10 /home/travis/miniconda3/envs/pandas-dev/bin/python
datapath = <function datapath.<locals>.deco at 0x7fd87cf76268>
    def test_legacy_sparse_warning(datapath):
        """
    
        Generated with
    
        >>> df = pd.DataFrame({"A": [1, 2, 3, 4], "B": [0, 0, 1, 1]}).to_sparse()
        >>> df.to_pickle("pandas/tests/io/data/pickle/sparseframe-0.20.3.pickle.gz",
        ...              compression="gzip")
    
        >>> s = df['B']
        >>> s.to_pickle("pandas/tests/io/data/pickle/sparseseries-0.20.3.pickle.gz",
        ...             compression="gzip")
        """
        with tm.assert_produces_warning(FutureWarning):
            simplefilter("ignore", DeprecationWarning)  # from boto
            pd.read_pickle(
                datapath("io", "data", "pickle", "sparseseries-0.20.3.pickle.gz"),
>               compression="gzip",
            )
pandas/tests/io/test_pickle.py:218: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <contextlib._GeneratorContextManager object at 0x7fd87ceed6a0>
type = None, value = None, traceback = None
    def __exit__(self, type, value, traceback):
        if type is None:
            try:
>               next(self.gen)
E               AssertionError: Caused unexpected warning(s): [('ResourceWarning', ResourceWarning("unclosed <socket.socket fd=18, family=AddressFamily.AF_INET, type=2049, proto=6, laddr=('10.20.0.67', 37224), raddr=('74.125.202.84', 443)>",), '/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/inspect.py', 732), ('ResourceWarning', ResourceWarning("unclosed <socket.socket fd=19, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('10.20.0.67', 46940), raddr=('172.217.214.95', 443)>",), '/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/inspect.py', 732)]
../../../miniconda3/envs/pandas-dev/lib/python3.6/contextlib.py:88: AssertionError
=============================== warnings summary ===============================
/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/fastparquet/dataframe.py:5
/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/fastparquet/dataframe.py:5
  /home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/fastparquet/dataframe.py:5: FutureWarning: pandas.core.index is deprecated and will be removed in a future version.  The public classes are available in the top-level namespace.
    from pandas.core.index import CategoricalIndex, RangeIndex, Index, MultiIndex
/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/statsmodels/tools/_testing.py:19
  /home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
    import pandas.util.testing as tm
```

--- a/pandas/tests/io/test_pickle.py
+++ b/pandas/tests/io/test_pickle.py
@@ -198,6 +198,7 @@ def test_pickle_path_localpath():
     tm.assert_frame_equal(df, result)
 
 
+@pytest.mark.xfail
 def test_legacy_sparse_warning(datapath):
     """
 

--- a/pandas/tests/io/test_pickle.py
+++ b/pandas/tests/io/test_pickle.py
@@ -198,7 +198,7 @@ def test_pickle_path_localpath():
     tm.assert_frame_equal(df, result)
 
 
-@pytest.mark.xfail
+@pytest.mark.xfail(reason="GitHub issue #31310", strict=False)
 def test_legacy_sparse_warning(datapath):
     """
 

